rest-service/manager_rest/upload_manager.py

```diff
@@ -697,8 +698,16 @@ def _prepare_and_process_doc(self,
             os.remove(archive_target_path)
             shutil.move(archive_name, archive_target_path)
             try:
-                wagon_target_path, _ = \
+                wagon_target_path, yamls_target_path = \
                     self._verify_archive(archive_target_path)
+                plugin_yaml, changes = _required_bc_changes(yamls_target_path)
+                if changes:
+                    bc_plugin_yaml = _make_bc_plugin_yaml(plugin_yaml, changes)
+                    _save_bc_plugin_yaml(
+                        os.path.join(archive_target_path, 'plugin_1_3.yaml'),
+                        bc_plugin_yaml
+                    )
+
```

Comment:
It seems the split of "find changes first, then pass them to the `make` function" just makes things harder, because now the "find" function needs to structure the paths...

Instead, I think it'd be easier if the interface was just a single function, that would update the yaml as soon as it found any incompatible items.  (and it would presumably have a flag `found = False` or something, so that it could do nothing if there's nothing to do)